# AprilTag 3 code parameters
# Find descriptions in apriltag/include/apriltag.h:struct apriltag_detector
#                      apriltag/include/apriltag.h:struct apriltag_family
tag_family:        'tag36h11' # options: tagStandard52h13, tagStandard41h12, tag36h11, tag25h9, tag16h5, tagCustom48h12, tagCircle21h7, tagCircle49h12
tag_threads:       2          # default: 2
tag_decimate:      1.0        # default: 1.0
tag_blur:          0.0        # default: 0.0
tag_refine_edges:  1          # default: 1
tag_debug:         0          # default: 0
max_hamming_dist:  2          # default: 2 (Tunable parameter with 2 being a good choice - values >=3 consume large amounts of memory. Choose the largest value possible.)
# Other parameters
publish_tf:        true       # default: false
z_up:              true
enable_detector:   false
transport_hint:    "raw"      # default: raw, see http://wiki.ros.org/image_transport#Known_Transport_Packages for options

# Syntax
# standalone_tags:
#   [
#     {id: ID, size: SIZE, name: NAME},
#     ...
#   ]
standalone_tags:
  [
    {id: 0, size: 0.042, name: tag_frame_0},
    {id: 1, size: 0.042, name: tag_frame_1},
    {id: 2, size: 0.042, name: tag_frame_2},
    {id: 3, size: 0.042, name: tag_frame_3},
    {id: 4, size: 0.042, name: tag_frame_4},
    {id: 5, size: 0.042, name: tag_frame_5},
    {id: 6, size: 0.042, name: tag_frame_6},
    {id: 7, size: 0.042, name: tag_frame_7},
    {id: 8, size: 0.042, name: tag_frame_8},
    {id: 9, size: 0.042, name: tag_frame_9},
    {id: 10, size: 0.042, name: tag_frame_10},
    {id: 11, size: 0.042, name: tag_frame_11},
    {id: 12, size: 0.042, name: tag_frame_12},
    {id: 13, size: 0.042, name: tag_frame_13},
    {id: 14, size: 0.042, name: tag_frame_14},
    {id: 15, size: 0.042, name: tag_frame_15},
    {id: 16, size: 0.042, name: tag_frame_16},
    {id: 17, size: 0.042, name: tag_frame_17},
    {id: 18, size: 0.042, name: tag_frame_18},
    {id: 19, size: 0.042, name: tag_frame_19},
    {id: 20, size: 0.042, name: tag_frame_20},
  ]
# Tag bundle definitions
# Remarks
# - name is optional
# - x, y, z have default values of 0 thus they are optional
# - qw has default value of 1 and qx, qy, qz have default values of 0 thus they are optional
#
# ### Syntax
#
# tag_bundles:
#   [
#     {
#       name: 'CUSTOM_BUNDLE_NAME',
#       layout:
#         [
#           {id: ID, size: SIZE, x: X_POS, y: Y_POS, z: Z_POS, qw: QUAT_W_VAL, qx: QUAT_X_VAL, qy: QUAT_Y_VAL, qz: QUAT_Z_VAL},
#           ...
#         ]
#     },
#     ...
#   ]
tag_bundles:
  [
  ]
